test/linux/mbox-server-tester.c

```diff
@@ -0,0 +1,113 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <sched.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "libmbox/mbox.h"
+#include "mailbox-map.h"
+#include "mbox-utils.h"
+
+#define PATH_SIZE 128
+
+#define RTPS_PATH_IN "/dev/mbox/1/mbox4"
+#define RTPS_PATH_OUT "/dev/mbox/1/mbox3"
+
+
+enum mbox_notif notif_type = MBOX_NOTIF_NONE;
+int timeout_ms = -1;
+
+static const char short_options[] = "n:t:c:h";
+static const struct option long_options[] = {
+    {"not-type",    required_argument,  NULL,   'n'},
+    {"timeout",     required_argument,  NULL,   't'},
+    {"cpu",         required_argument,  NULL,   'c'},
+    {"help",        no_argument,        NULL,   'h'},
+    {0, 0, 0, 0}
+};
+
+int main(int argc, char** argv)
+{
+	struct mbox mbox_in;
+	struct mbox mbox_out;
+
+	//char devpath_out_buf[PATH_SIZE];
+	//const char *devpath_out = "4";
+	//devpath_out = expand_path(devpath_out, devpath_out_buf, sizeof(devpath_out_buf));
+
+	int cpu = -1;
+
+
+	int c;	
+	while ((c = getopt_long(argc, argv, short_options, long_options, NULL)) != -1) {
+        switch (c) {
+        case 'n':
+            if (!strcmp(optarg, "none")) {
+                notif_type = MBOX_NOTIF_NONE;
+            } else if (!strcmp(optarg, "select")) {
+                notif_type = MBOX_NOTIF_SELECT;
+            } else if (!strcmp(optarg, "poll")) {
+                notif_type = MBOX_NOTIF_POLL;
+            } else if (!strcmp(optarg, "epoll")) {
+                notif_type = MBOX_NOTIF_EPOLL;
+            } else {
+                fprintf(stderr, "Unknown notification type: %s\n", optarg);
+                //usage(argv[0], EINVAL);
+            }
+            break;
+        case 't':
+            timeout_ms = atoi(optarg);
+            break;
+        case 'c':
+            cpu = atoi(optarg);
+            break;
+        // case 'h':
+        //     usage(argv[0], 0);
+        //     break;
+        // default:
+        //     usage(argv[0], EINVAL);
+        //     break;
+        // }
+    	}
+	}
+
+	notif_type = MBOX_NOTIF_POLL;
+	//mbox_open_or_die(&mbox_in, devpath_out, notif_type, timeout_ms, O_RDONLY);
+	mbox_open_or_die(&mbox_in, RTPS_PATH_IN, notif_type, timeout_ms, O_RDONLY);
+	mbox_open_or_die(&mbox_out, RTPS_PATH_OUT, notif_type, timeout_ms, O_RDWR);
+
+	//TO DO: PIN TO CPU IF USER SET CPU
```

Comment:
Please add in the pin CPU functionality.  Take a look at the "CPU_SET" call in mboxtester.c for an example.